src/components/Input.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react';
+
+// Example:
+// <Input name="email" type="email" label="Email Address"
+// required={true} validation="/^\w+@[a-zA-Z_]+?\.[a-zA-Z]{2,3}$/" />
+
+class Input extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      inputText: "",
+      validated: true
+    }
+    this.setInputText = this.setInputText.bind(this);
+    this.checkValidation = this.checkValidation.bind(this);
+  }
+
+  setInputText(e) {
+    e.preventDefault();
```

Comment:
this will not actually do anything in React as events here aren't native but synthetic. 

See here for more info: https://stackoverflow.com/questions/36316846/react-onclick-and-preventdefault-link-refresh-redirect

Reply:
I thought this was necessary for maintaining a controlled component.  Is it because the call in the input prop is `onChange={this.setInputText}` instead of `onChange={ e => this.setInputText(e) }`?